src/nb_func.cpp

```diff
@@ -563,8 +561,15 @@ static PyObject *nb_func_vectorcall_complex(PyObject *self,
                        has_var_args   = f->flags & (uint32_t) func_flags::has_var_args,
                        has_var_kwargs = f->flags & (uint32_t) func_flags::has_var_kwargs;
 
-            /// Number of positional arguments
-            size_t nargs_pos = f->nargs - has_var_args - has_var_kwargs;
+            // Number of C++ arguments eligible to be filled from individual
+            // Python positional arguments
+            size_t nargs_pos = f->nargs_pos;
+
+            // Number of C++ arguments eligible to be filled from individual
+            // Python arguments (either positional or keyword), except that:
+            // if has_var_args, then the one at index nargs_pos is a nb::args
+            // that we'll fill in afterward.
+            size_t nargs_step1 = f->nargs - has_var_kwargs;
```

Comment:
Not sure I understand. Your comment mentions `has_var_args`, and that explanation made sense to me. But the code then uses `has_var_kwargs`.

Reply:
Reworded for more clarity. A C++ parameter that takes variadic kwargs is excluded from this count. A C++ parameter that takes variadic positional args is included in the count but then skipped in the loop below. Rationale: if a nb::args parameter is present, it's between the pos-or-kw args and the kw-only args; it's better to deal with both of those categories in the same loop below than to duplicate the code for each type.

Reply:
Awesome.